llvm/lib/Target/AMDGPU/VOP3Instructions.td

```diff
@@ -156,10 +156,12 @@ defm V_MAX_F64 : VOP3Inst <"v_max_f64", VOP3_Profile<VOP_F64_F64_F64>, fmaxnum_l
 } // End SchedRW = [WriteDoubleAdd]
 
 let SchedRW = [WriteIntMul] in {
+defm V_MUL_LO_I32 : VOP3Inst <"v_mul_lo_i32", V_MUL_PROF<VOP_I32_I32_I32>>;
```

Comment:
You don't need to handle V_MUL_LO_I32 differently from the other mul opcodes here. It was never generated because its behaviour was always identical to the U32 version, so recent generations have removed support for it, so it does not matter if you set the IsInvalid flag on it.

Reply:
That's good to know. I've changed the code back to handle all the mul opcodes together.